src/Features/Core/Portable/CodeFixesAndRefactorings/FeaturesFixAllGetFixesService.cs

```diff
@@ -5,20 +5,17 @@
 using System;
 using System.Composition;
 using Microsoft.CodeAnalysis.CodeFixesAndRefactorings;
-using Microsoft.CodeAnalysis.Host;
 using Microsoft.CodeAnalysis.Host.Mef;
 
 namespace Microsoft.CodeAnalysis.Editor.Implementation.Suggestions;
 
-[ExportWorkspaceServiceFactory(typeof(IFixAllGetFixesService), ServiceLayer.Host), Shared]
-internal sealed class FeaturesFixAllGetFixesService : AbstractFixAllGetFixesService, IWorkspaceServiceFactory
+[ExportWorkspaceService(typeof(IFixAllGetFixesService), ServiceLayer.Default), Shared]
+[method: ImportingConstructor]
+[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+internal sealed class FeaturesFixAllGetFixesService() : AbstractFixAllGetFixesService
 {
-    [ImportingConstructor]
-    [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-    public FeaturesFixAllGetFixesService()
+    protected override Solution? GetChangedSolution(Workspace workspace, Solution currentSolution, Solution newSolution, string fixAllPreviewChangesTitle, string fixAllTopLevelHeader, Glyph glyph)
```

Comment:
😕 Why override the base method instead of changing the base method to return `newSolution`?

Reply:
base method is abstract.  my preference is to make the leaves pick their particular chocie on what to to.  I'm not a fan of default behavior that may be missed when subtyping classes.  But it's a personal preference thing here.  Ultimately, it would work out the same in terms of how this runs in practice.